microovn/frr/bgp/redirect.go

```diff
@@ -238,28 +284,157 @@ func redirectBgp(ctx context.Context, s state.State, extConnections []types.BgpE
+// the BGP redirect. This includes:
+//   - Logical Router
+//   - Logical Switches
+//   - OVS external bridges
+//   - OVS ports
+//   - OVN bridge mappings
+//
+// Other OVN resources remain untouched.
+func teardownAll(ctx context.Context, s state.State) error {
+	var allErrors error
+	// Find and remove Logical Router used for BGP redirect
+	logicalRouters, err := ovnCmd.NBCtlCluster(ctx, "--bare", "--columns", "name",
+		"find", "logical_router", fmt.Sprintf("external-ids:%s=true", BgpManagedTag),
+	)
+	if err != nil {
+		allErrors = errors.Join(allErrors, fmt.Errorf("failed to lookup Logical Routers managed by MicroOVN: %v", err))
+	} else {
+		for _, logicalRouter := range parseOvnFind(logicalRouters) {
+			_, err = ovnCmd.NBCtlCluster(ctx, "lr-del", logicalRouter)
+			if err != nil {
+				allErrors = errors.Join(allErrors, fmt.Errorf("failed to delete Logical Router '%s': %v", logicalRouter, err))
+			}
+		}
+	}
+
+	// Find and remove Logical Switches used to connect to external networks
+	logicalSwitches, err := ovnCmd.NBCtlCluster(ctx, "--bare", "--columns", "name",
+		"find", "logical_switch", fmt.Sprintf("external-ids:%s=true", BgpManagedTag),
+	)
+	if err != nil {
+		allErrors = errors.Join(allErrors, fmt.Errorf("failed to lookup Logical Switches managed by MicroOVN: %v", err))
+	} else {
+		for _, logicalSwitch := range parseOvnFind(logicalSwitches) {
+			_, err = ovnCmd.NBCtlCluster(ctx, "ls-del", logicalSwitch)
+			if err != nil {
+				allErrors = errors.Join(allErrors, fmt.Errorf("failed to delete Logical Switch '%s': %v", logicalSwitch, err))
+			}
+		}
+	}
+
+	// Find and remove external OVS bridges
+	bridges, err := ovnCmd.VSCtl(ctx, s, "--bare", "--columns", "name",
+		"find", "bridge", fmt.Sprintf("external-ids:%s=true", BgpManagedTag),
+	)
+	if err != nil {
+		allErrors = errors.Join(allErrors, fmt.Errorf("failed to lookup OVS Bridges managed by MicroOVN: %v", err))
+	} else {
+		for _, bridge := range parseOvnFind(bridges) {
+			_, err = ovnCmd.VSCtl(ctx, s, "del-br", bridge)
+			if err != nil {
+				allErrors = errors.Join(allErrors, fmt.Errorf("failed to delete OVS Bridge '%s': %v", bridge, err))
+			}
+		}
+	}
+
+	// Find and remove OVS ports used for BGP redirect
+	ports, err := ovnCmd.VSCtl(ctx, s, "--bare", "--columns", "name",
+		"find", "port", fmt.Sprintf("external-ids:%s=true", BgpManagedTag),
+	)
+	if err != nil {
+		allErrors = errors.Join(allErrors, fmt.Errorf("failed to lookup OVS Ports managed by MicroOVN: %v", err))
+	} else {
+		for _, port := range parseOvnFind(ports) {
+			_, err = ovnCmd.VSCtl(ctx, s, "del-port", port)
+			if err != nil {
+				allErrors = errors.Join(allErrors, fmt.Errorf("failed to delete OVS Port '%s': %v", port, err))
+			}
 		}
+	}
+
+	// Cleanup ovn-bridge mappings for external networks
+	ovnBridgeMapping, err := ovnCmd.VSCtl(ctx, s, "get", "Open_vSwitch", ".", "external-ids:ovn-bridge-mappings")
```

Comment:
This call creates an error when it does not exist, which is the case when a user has ran `microovn enable bgp` without options only to then do `microovn disable bgp`.